packages/core/src/m3/builtins.ts

```diff
@@ -83,19 +83,44 @@ const builtinFeatures = {
     inamed_name
 }
 
-export class DefaultPrimitiveTypeDeserializer implements PrimitiveTypeDeserializer {
 
-    private deserializerByType = new Map<Datatype, SpecificPrimitiveTypeDeserializer>()
+/**
+ * Determines whether two datatypes should be structurally equal based on equality of: meta type, key, and language's key.
```

Comment:
Why calling the method "should"? I would expect it to do behave as an assert method and through an exception if the two elements are not equals.
I would find a method name like `areIdentical` more obvious

Reply:
Because two instance of a `Datatype` **should** be equal if their metadata (metapointers and metatypes) are identical. I could also compare them really structurally, but then their metadata should also be identical. And I wouldn't know how to interpret two datatypes that have nonidentical metadata but are otherwise structurally the same, in this context.